### ISSUE TITLE
Flag to ignore certain error types

### ISSUE THREAD

**Issue by connorshea** [AaronC81/sord#57]
**Is your feature request related to a problem? Please describe.**
I'd like to run Sord and only have comments for `SORD_ERROR`s and other `warn`-level errors. Right now there are a lot of messages about OMITs, but I only care about WARNs.

There's a lot of stuff like this that isn't very useful for me:

```ruby
# sord omit - no YARD type given for "gem_names", using T.untyped
# sord omit - no YARD return type given, using T.untyped
sig { params(gem_names: T.untyped).returns(T.untyped) }
def dependency_specs(gem_names); end

# sord omit - no YARD type given for "gem_names", using T.untyped
# sord omit - no YARD return type given, using T.untyped
sig { params(gem_names: T.untyped).returns(T.untyped) }
def unmarshalled_dep_gems(gem_names); end
```

**Describe the solution you'd like**
There are three main ideas I have for implementing this:

- Exclude messages of certain types: `sord test.rbi --exclude-messages=omit,duck`
- Include messages of certain types: `sord test.rbi --include-messages=warn`
- Error level: `sord test.rbi --error-level=warn`

The first two could both be implemented, if we really want to.

The 'error level' flag would require that we decide what the 'level's of each error type are, and since we have omit / duck / infer / warn, that wouldn't really make sense.

**Additional context**
I assume the flags would effect both the logging _and_ code comments, having two separate flags for the logging and the code comments would unnecessarily complicate things.

**Comment by AaronC81** [comment]
Fantastic idea; this could make Sord's output much less noisy.

I'd personally be in favour of implementing both `--exclude-messages` and `--include-message`, as a sort of blacklist/whitelist system respectively.

In terms of how specifying these should work, there are two ways I can think of. Either:

  - Specifying multiple logging types by comma-separating them, like you've done in your example: `sord test.rbi --exclude-messages=omit,duck`, or
  - Specifying the same option multiple times for each logging type: `sord test.rbi --exclude-message omit --exclude-message duck` 

I'm not particularly sure which I prefer - what are your thoughts?

**Comment by connorshea** [comment]
I definitely prefer the comma-separated option out of those two, but honestly you should go with whatever is easier to implement :)

**Comment by AaronC81** [comment]
Alrighty - thinking about it I'm not sure if Commander even supports specifying the same argument multiple times, so I'll go with the first one.

**Comment by AaronC81** [comment]
Closed by #64!
